rmf_traffic_ros2/src/rmf_traffic_ros2/schedule/MirrorManager.cpp

```diff
@@ -180,15 +244,88 @@ class MirrorManager::Implementation
       request.version = 0;
       request.full_update = true;
     }
-    request_changes_pub->publish(request);
+    request_changes_client->async_send_request(
+      std::make_shared<RequestChanges::Request>(request),
+      [&](const RequestChangesFuture response)
+      {
+        // Check how the schedule node handled the request. The actual queries
+        // update will come separately over the query update topic; this is
+        // just whether the request was handled successfully or not.
+        auto value = *response.get();
+        if (value.result == RequestChanges::Response::UNKNOWN_QUERY_ID)
+        {
+          redo_query_registration();
+        }
+      });
+  }
+
+  void redo_query_registration()
+  {
+    RCLCPP_DEBUG(node.get_logger(), "Redoing query registration");
+    // Make sure nothing is truly coming in on this topic and triggering a
+    // callback while we are remaking it
+    mirror_update_sub.reset();
+
+    register_query_client =
+      node.create_client<RegisterQuery>(RegisterQueryServiceName);
+    redo_query_registration_timer = node.create_wall_timer(
+      100ms,
+      std::bind(
+        &MirrorManager::Implementation::redo_query_registration_callback,
+        this));
+  }
+
+  void redo_query_registration_callback()
+  {
+    if (register_query_client->wait_for_service(
+          std::chrono::milliseconds(100)))
```

Comment:
Since we're already waiting 100ms between calls to this function, let's not wait at all here. Let's just check if the service is available and move along if it's not. Otherwise we'll be needlessly holding up other operations that need to happen on this hot event loop.

I'll go ahead and remove this.